Mathlib/LinearAlgebra/TensorProduct/Quotient.lean

```diff
@@ -0,0 +1,241 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, Jujian Zhang
+-/
+
+import Mathlib.RingTheory.Ideal.Operations
+import Mathlib.LinearAlgebra.TensorProduct.Basic
+import Mathlib.LinearAlgebra.Quotient
+import Mathlib.LinearAlgebra.Prod
+import Mathlib.RingTheory.Ideal.Quotient
+
+/-!
+
+# Interaction between Quotients and Tensor Products
+
+This file contains constructions that relate quotients and tensor products.
+Let `M, N` be `R`-modules, `m ≤ M` and `n ≤ N` be an `R`-submodules and `I ≤ R` an ideal. We prove
+the following isomorphisms:
+
+## Main results
+- `TensorProduct.quotientTensorQuotientEquiv`:
+  `(M ⧸ m) ⊗[R] (N ⧸ n) ≃ₗ[R] (M ⊗[R] N) ⧸ (m ⊗ N ⊔ M ⊗ n)`
+- `TensorProduct.quotientTensorEquiv`:
+  `(M ⧸ m) ⊗[R] N ≃ₗ[R] (M ⊗[R] N) ⧸ (m ⊗ N)`
+- `TensorProduct.tensorQuotientEquiv`:
+  `M ⊗[R] (N ⧸ n) ≃ₗ[R] (M ⊗[R] N) ⧸ (M ⊗ n)`
+- `TensorProduct.quotTensorEquivQuotSMul`:
+  `(R ⧸ I) ⊗[R] M ≃ₗ[R] M ⧸ (I • M)`
+- `TensorProduct.tensorQuotEquivQuotSMul`:
+  `M ⊗[R] (R ⧸ I) ≃ₗ[R] M ⧸ (I • M)`
```

Comment:
This file looks good. Please go ahead and split it into a new PR.

Reply:
Thanks. See #17803